cmd/rook/ceph/osd.go

```diff
@@ -251,8 +259,25 @@ func prepareOSD(cmd *cobra.Command, args []string) error {
 	clusterInfo.OwnerInfo = ownerInfo
 	clusterInfo.Context = cmd.Context()
 	kv := k8sutil.NewConfigMapKVStore(clusterInfo.Namespace, context.Clientset, ownerInfo)
+
+	// destroy the OSD using the OSD ID
+	var replaceOSD *osd.OSDReplaceInfo
+	if replaceOSDID != -1 {
+		osdInfo, err := osddaemon.GetOSDInfoById(context, &clusterInfo, replaceOSDID)
```

Comment:
Can this be moved inside the `destroyOSD()` method? To keep everything about the destroy in a single place.